src/deadline/client/cli/_groups/bundle_group.py

```diff
@@ -218,6 +230,12 @@ def _decide_cancel_submission(upload_group: AssetUploadGroup) -> bool:
             submitter_name=submitter_name,
         )
 
+        # Reset the storage profile ID so that we don't save it to disk.
+        # Note: The config object is a singleton, so the config object that
+        # we may have modified above if a storage_profile_id was given is saved
+        # into that singleton. Setting the job_id writes that singleton to disk
+        set_setting(storage_profile_config_name, prev_storage_id, config=config)
```

Comment:
so I'm understanding correctly, the config is being passed in both so nothing is saved to disk when this particular call happens. But due to the singleton nature of the config object, setting the job id would trigger setting this to disk. Correct?

Reply:
Yes, that is 100% correct.